src/main/java/net/apnic/rdap/rdap/RDAPError.java

```diff
@@ -0,0 +1,98 @@
+package net.apnic.rdap.rdap;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class represent an error response in the RDAP protocol.
+ *
+ * @see https://tools.ietf.org/html/rfc7483
+ */
+public class RDAPError
+    extends RDAPObject
+{
+    private List<String> description = null;
```

Comment:
I'd make the trade-off for immutability. (and there is no such thing as a perfect design, design is all about making trade-offs).

You could keep your factory by just asking for all the inputs up front - loosing the generic approach by adding a ```createRdapError(...)``` would be the most straight forward.

But I think explicitly putting RDAPDefaults into the constructor is the most cohesive approach - since it tells the reader that RDAPDefaults are actually needed to create a valid instance.